extensions/narayana-jta/runtime/src/main/java/io/quarkus/narayana/jta/runtime/TransactionManagerConfiguration.java

```diff
@@ -29,7 +29,7 @@ public final class TransactionManagerConfiguration {
      * If the value is not absolute then the directory is relative
      * to the <em>user.dir</em> system property.
      */
-    @ConfigItem(defaultValue = "ObjectStore")
+    @ConfigItem(defaultValue = "target/ObjectStore")
```

Comment:
TBH, I'm not exactly sure how to solve this problem. target is Maven specific and related to the build. It makes no sense to use if for let's say a Gradle build or even running in production mode.

@aloubyansky are you aware of somewhere we had to solve the same issue? Or do you have an idea on where we could store this without it being an issue, neither for the build of projects nor for running projects in production?

Reply:
`CurateOutcomeBuildItem.getApplicationModel().getAppArtifact().getWorkspaceModule().getBuildDir()`
Theoretically, `getWorkspaceModule()` could return `null` in case we are building an app from a jar instead of project, in which case you could simply use `Path.of("")` or `Path.of("target"), making sure it exists.

Reply:
Yeah, my problem is that we would need a different behavior during build/test and actual run where creating the directory where the binary is could make sense (but I'm not sure it's permitted in our Docker images for instance).

Reply:
There is also `OutputTargetBuildItem`

Reply:
Sure but that doesn't solve the problem of how to make the configuration different for test/build and for actual run. And where we should store this on run.

@cescoffier do we store some Vert.x data somewhere already? Some that are not considered temporary data?